packages/legacy/src/Plans/components/Wizard/GeneralForm.tsx

```diff
@@ -50,10 +52,21 @@ export const GeneralForm: React.FunctionComponent<IGeneralFormProps> = ({
   const inventoryProvidersQuery = useInventoryProvidersQuery();
   const clusterProvidersQuery = useClusterProvidersQuery(namespace);
   const namespacesQuery = useNamespacesQuery(form.values.targetProvider);
+  const secretsQuery = useSecretsForProvidersQuery(clusterProvidersQuery, namespace);
+
   const targetNsFoundInQueriesNs = !!namespacesQuery.data?.find(
     (namespace) => form.values.targetNamespace === namespace.name
   );
 
+  const [sourceClusterProvider, setSourceClusterProvider] = React.useState<IProviderObject>(); // TODO: Use a default for edit form
```

Comment:
Because the form field actually holds a `InventoryProvider` not the `IProviderObject` clusterProvider.  The name of the secret is only on the cluster provider.